src/editor/Collaboration.js

```diff
@@ -59,7 +61,14 @@ define(function (require, exports, module) {
             var rootDir = StartupState.project("root");
             if(added) {
                 added.forEach(function(addedFile) {
-                    _webrtc.sendToAll("file-added", {path: Path.relative(rootDir, addedFile.fullPath), isFolder: addedFile.isDirectory});
+                    _webrtc.getPeers().forEach(function(peer) {
+                        if (!_received[Path.relative(StartupState.project("root"), addedFile._path)]) {
```

Comment:
store `Path.relative(StartupState.project("root"), addedFile._path)` in a variable above this `forEach` loop so that you don't evaluate it every time.